### DISCUSSION TITLE
Restricted amount of Tags ?

### DISCUSSION THREAD

**Discussion by lotfibkr-b** [RudolphRiedel/FT800-FT813#119] [General]
Hello, 

I have been using your library since October 2022, and it has been really useful for my touchscreen projects ever since !

However, in my actual project, the touchscreen requires quite a lot of buttons (each of them is assigned a tag value ofc), and I have reached case 255.
But from that point, no more button with a tag higher than 255 wants to "work".

I looked up the function associated with the tag reading value (EVE_memRead8) and noticed two other similar functions (EVE_memRead16 & EVE_memRead32) which looked promising to solve the issue. 

Unfortunately, even when using those functions (and at the same time changing the length of the "tag" variable from uint8_t to uint16_t or uint32_t) the buttons are still not working.

Do you have any idea on how to solve this "issue"? 

Thank you very much in advance!

**Comment by RudolphRiedel** [comment]
The tags "only" go from 0 to 255, so directly there is no way around this.
Are you really trying to use more than 255 buttons on a single screen?

Apart from using per-screen tag values, one way out could be to use groups of buttons sharing a tag and then using the touch coordinates to find the specific button that was touched within a group.

**Reply by lotfibkr-b** [reply]
First, thank you for your fast answer! 

Well, when you say "Are you really trying to use more than 255 buttons **on a single screen**?", do you mean one touchscreen or one window ? And by "window", I mean a TFT_display() function.

To elaborate my situation a bit further, I've created around 40-50 TFT_display() "windows" (each of them has a specific task) and there are between 1 and 10 buttons on each of those windows. 
And now I have to create some new windows with new buttons on them, but that is where I reached the maximum amount of tags for the buttons.

But from what you are saying, I spot that the solution might come from "using per-screen tag values". But that is not how I structured my code so far. 

Do you think that you have a code example using that method, so I could try to use it and modify my actual code?

**Reply by RudolphRiedel** [reply]
I just checked and the last time I was working with more than one screen/window/view, I did not ran out of tags.
I used a screen_main() function and a screen_config() function.
And the screen_config() function with 11 buttons was actually shared between four options. Using a "screen" variable did not only determined what screen was displayed but also which dataset was manipulated and which dataset to display from.

**Reply by lotfibkr-b** [reply]
Am I right when I say that you had to assign one case of your switch case to each of your 11 buttons?

And what is the point of your "screen_config()" function?

**Comment by RudolphRrr** [comment]
>Am I right when I say that you had to assign one case of your switch case to each of your 11 buttons?

In this project I did it like this, yes.
It was not really complex overall.
Well, to the user this still looked like 4 different screens with the same 11 buttons and same layout, so my application kind of shared these buttons by using a screen number as additional information.

>And what is the point of your "screen_config()" function?

That is just the name of the function that provided the part of the display list that was needed for the screen "config".
I used TFT_display() like in my example code, including a headline and a logo added with CMD_APPEND.

Then I used switch(screen) and called the necessary function.
And the end of the display list was back TFT_display() as usual.

So I named it "screen", but the way I used my functions was more like as a window.
